File: Sources/CodexTaskMonitorCore/MonitorListUpdate.swift
public enum MonitorListUpdate {
    public static func insertedID(from oldIDs: [String], to newIDs: [String]) -> String? {
        let oldIDs = Set(oldIDs)
        guard oldIDs.isSubset(of: Set(newIDs)) else { return nil }
        return newIDs.first { !oldIDs.contains($0) }
    }
}

File: windows/CodexTaskMonitor.Core/MonitorListUpdate.cs
namespace CodexTaskMonitor.Core;

public static class MonitorListUpdate
{
    public static string? InsertedId(IReadOnlyList<string> oldIds, IReadOnlyList<string> newIds)
    {
        var removed = oldIds.Except(newIds, StringComparer.Ordinal).Any();
        var inserted = newIds.Except(oldIds, StringComparer.Ordinal).ToArray();
        return !removed && inserted.Length == 1 ? inserted[0] : null;
    }
}
